### PR TITLE
Add custom webplayer for javascript users

### PR DESCRIPTION
**Pull request by AlexFence**
Okay I Implemented Issue https://github.com/barrucadu/lainonlife/issues/3. 
I started at like 4am, and the Issue updated so yeah now its missing the progressbar.
But I thought I just open the pull request anyway, because most of it is already done. I hope that is okay, never opened a pull request before, sorry in advance.

My commit includes:
 * play and pause button
 * mute button
 * volume slider

### CODE REVIEW

**Comment by barrucadu** [comment]
Looking pretty good, just some feedback:

- The `.scss` and `.less` files aren't needed, as those are used to produce the eventual `.css` files.
- `font-awesome.css` file isn't needed either, as only `font-awesome.min.css` is included in the page
- `FontAwesome.otf` isn't needed either, as the css only pulls in the `ttf`, `woff`, `woff2`, and `svg` variants.

The diff should be a bit less huge with those unnecessary files gone.

Do you feel up to implementing the progress bar as well?

File: player.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        const newPlayer = document.createElement("div")
+        newPlayer.setAttribute("id", "lainplayer");
+        newPlayer.innerHTML = playerHTML;
+        // hide old player and add the new one
+        oldPlayer.setAttribute("style", "display:none;");
+        oldPlayer.after(newPlayer);
+
+    }
+
+    function cangeSource(source) {
```
Comment (barrucadu) [inline player.js]: This should be `changeSource`.

File: player.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        console.log("fug");
```
Comment (barrucadu) [inline player.js]: This is unnecessary.

File: player.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
+const LainPlayer = (() => {
+    const playerHTML = "<a id='play-toggle' onclick='LainPlayer.play()'><i class='fa fa-play' aria-hidden='true'></i></a><i id=volume-button class='fa fa-volume-up' aria-hidden='true' onclick='LainPlayer.toggleMute()'></i><input type='range' value=0.51 min=0 max=1  step=0.01 oninput='LainPlayer.changeVolume(this.value)'/>"
```
Comment (barrucadu) [inline player.js]: The current conditional stuff uses the `withscript` and `noscript` classes, is there some reason to prefer including the HTML in the javascript here?

File: player.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        // Load the new audio stream.
+        audioTag.load();
+
+        // Start playing, if it was before.
+        if(!paused) {
+            audioTag.play();
+        }
+    }
+
+    function updatePlayButton() {
```
Comment (barrucadu) [inline player.js]: This function should just modify the class list of the button, rather than alter its HTML.

**Comment by AlexFence** [comment]
First: thanks for the constructive feedback, not really used to that.

Yeah, I would like to do the progressbar as well, it should display the current progress of the playing song right? Which differs from the default audio tag.

I'll also change the requested stuff

**Comment by barrucadu** [comment]
I've had a better idea for the fontawesome stuff: pull it in as a git submodule, rather than including the files directly.

>Yeah, I would like to do the progressbar as well, it should display the current progress of the playing song right? Which differs from the default audio tag.

Yes.  If you go to, eg, https://lainon.life/playlist/everything.json you'll get an object like so:

```json
{ "before": [ ... ]
, "current": { "time": "267", ... }
, "after": [ ...]
, "elapsed": "210.094"
}
```

File: player.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            volumebtn.classList.remove('fa-volume-up');
+            volumebtn.classList.remove('fa-volume-off');
+            volumebtn.classList.add('fa-volume-down');
+        }
+    }
+
+    function toggleMute() {
+        if (!muted) {
+            audioTag.volume = 0;
+            document.getElementById("lainplayer").children[2].value = 0;
```
Comment (barrucadu) [inline player.js]: This is really fragile, use an id.

**Comment by barrucadu** [comment]
If I start the radio playing and then try to change channel, I get this in the console:

```
Uncaught TypeError: Assignment to constant variable.
    at getCurrentTime (player.js:80)
    at updateProgress (player.js:89)
    at Object.updateProgress (player.js:105)
    at radio.js:126
    at XMLHttpRequest.httpRequest.onreadystatechange (radio.js:9)
```

**Comment by barrucadu** [comment]
Alright, looking good.

Just one final feature request: the progress bar should smoothly scroll. The `updateProgress` function should still set its absolute progress, but in between updates it should smoothly move as well.

**Comment by AlexFence** [comment]
Okay, I'll look into it in the evening.

**Comment by barrucadu** [comment]
Great! Thanks for implementing this and putting up with my constant demands.

**Pull request merged**